src/App.tsx

```diff
@@ -1,223 +1,218 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+  useMemo,
+  useCallback,
+} from 'react';
+
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  getTodos, deleteTodo, addTodo, updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { ErrorNotification } from
+  './components/ErrorNotification/ErrorNotification';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { FilterType } from './types/FilterType';
+import { ErrorType } from './types/ErrorType';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterType, setFilterType] = useState(FilterType.ALL);
+  const [isAddingTodo, setIsAddingTodo] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [title, setTitle] = useState('');
+  const [selectedTodosId, setSelectedTodosId] = useState<number []>([]);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
```

Comment:
Try to use something like this for `TodoInfo`. Your Todo is not focused when I double click it. I need to click third time to make it focused. Due to this problem, it is more difficult to use onBlur and 'Escape'